run-make-check.sh

```diff
@@ -64,12 +64,21 @@ function run() {
     if test -f ./install-deps.sh ; then
 	$DRY_RUN ./install-deps.sh || return 1
     fi
-    $DRY_RUN ./autogen.sh || return 1
-    $DRY_RUN ./configure "$@"  --with-librocksdb-static --disable-static --with-radosgw --with-debug --without-lttng \
-        CC="ccache gcc" CXX="ccache g++" CFLAGS="-Wall -g" CXXFLAGS="-Wall -g" || return 1
-    $DRY_RUN make $BUILD_MAKEOPTS || return 1
-    $DRY_RUN make $CHECK_MAKEOPTS check || return 1
-    $DRY_RUN make dist || return 1
+    export TMPDIR=$(mktemp -d -p ${TMPDIR:-/tmp} -u ceph.XXX)
+    if test -x ./do_cmake.sh ; then
+        $DRY_RUN ./do_cmake.sh || return 1
+        cd build
+        export CTEST_OUTPUT_ON_FAILURE=1
+        $DRY_RUN make $BUILD_MAKEOPTS check || return 1
```

Comment:
and to address Marcus' issue regarding to "why we install stuff into /tmp" with "make all", i removed ceph-disk and ceph-detect-init from ALL. so we need to use "make check", but "make check" runs "ctest" also, so i replaced "make all" + "ctest" with "make check". i will add "CTEST_PARALLEL_LEVEL" to make it parallel.